CRM/Gpapi/Processor.php

```diff
@@ -457,4 +457,50 @@ public static function setContactIdByHash(&$contact_data) {
 
     return TRUE;
   }
+
+  /**
+   * Update contact.
+   * Set param contact id if it is found by contact 'bpk_extern'
+   *
+   * @param $contact_data
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public static function updateContactByBpk(&$contact_data) {
+    $bpk_extern_field_id = civicrm_api3('CustomField', 'getvalue', [
+      'return' => 'id',
+      'custom_group_id' => 'bpk',
+      'name' => 'bpk_extern',
+    ]);
+    $name = 'custom_' . $bpk_extern_field_id;
+    if (empty($contact_data[$name])) {
```

Comment:
The parameter name for BPK in `Engage.signpetition` is `bpk`, rather than `custom_xxx`. (Same in line 482.)